enhancements/hypershift/support-disabling-image-registry-via-capability.md

```diff
@@ -0,0 +1,573 @@
+is disabled
+* OpenShift Engineers are able to extend the list of capabilities available
+in Hypershift, allowing other optional components to be disabled in the future
+
+### Non-Goals
+
+* Supporting the full set of OpenShift cluster capabilities
+* Moving management of hosted control plane components to CVO
+([OTA-951](https://issues.redhat.com/browse/OTA-951))
+
+## Proposal
+
+`HostedCluster` grows an optional slice field `disabledCapabilities` in its
+`spec`, indicating what components users want to disable at install time.
+
+Initially, the only supported value in this slice is `ImageRegistry`.
+
+Editing this field at runtime is only allowed if the user is removing
+capabilities from the slice. Adding capabilities to the slice is not supported.
+In other words, disabling capabilities at runtime is not supported.
+
+The `control-plane-operator` is responsible for reconciling both, the Image
+Registry and Cluster Version Operator. Both reconcilers will need to change.
+
+The Image Registry reconciler needs to look for an entry matching the
+`ImageRegistry` capability in the `HostedCluster.spec.disabledCapabilities`
+field, and skip reconciliation when the `ImageRegistry` capability is found.
+
+The `CVO` reconciler needs to edit the `ClusterVersion` resource, setting
+`spec.capabilities.baselineCapabilitySet` to `"None"`, then calculating the
+difference between `ClusterVersionCapabilitySets[ClusterVersionCapabilitySetCurrent]`
+and `HostedCluster.spec.disabledCapabilities`, assigning the resulting set to
+field `spec.capabilities.additionalEnabledCapabilities`.
+
+The `hypershift` CLI grows a `--disable-cluster-capabilities` option. Initially
+this option will only accept a single value: `ImageRegistry`. In the future it
+can grow to support a comma separated list of cluster capabilities.
+
+### Workflow Description
+
+1. Cluster service consumer creates a hosted cluster via `hypershift` CLI, setting
+`--disable-cluster-capabilities=ImageRegistry`
+1. The `hypershift` CLI creates a `HostedCluster` resource, setting
+`.spec.disabledCapabilities` to `["ImageRegistry"]` as requested by the user
+1. Hypershift control-plane-operator detects the ImageRegistry capability is
+disabled in the `HostedCluster`, and skips creating resources for the Image
+Registry operator, ensuring the `ClusterVersion` resource reflects this
+1. The guest cluster becomes available and functional without the image registry
+1. The hosted control-plane becomes available and functional without the
+cluster image registry operator running
+1. Image Registry operator and operand resources are not present in guest or
+management clusters
+1. Image Registry infrastructure resources are not created in cloud provider
+(i.e. storage bucket, IAM user, etc)
+1. No pull-secrets are created for any service accounts, regardless of when
+the service account is created (install time or runtime)
+
+### API Extensions
+
+```go
```

Comment:
fyi, I've already created the API changes in https://github.com/openshift/hypershift/pull/5415